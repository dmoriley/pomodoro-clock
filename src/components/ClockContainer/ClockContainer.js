import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './ClockContainer.module.scss';
import { Clock, ClockActions, ClockSetting } from '../index';

const ClockContainer = props => {

  const [breakLength, setBreakLength] = useState(5);
  const [sessionLength, setSessionLength] = useState(20);

  // handle play should change the border color of the clock to red
  // create and import function that formats milliseconds into the format mm:ss
  return (
    <div className={styles.root}>
      <div className={styles.settings}>
        <ClockSetting 
          title="break length" 
          setting={breakLength} 
          handleDecrement={() => setBreakLength(breakLength !== 0 ? breakLength - 1 : 0)}
          handleIncrement={() => setBreakLength(breakLength + 1)}
        />
        <ClockSetting 
          title="session length"
          setting={sessionLength} 
          handleDecrement={() => setSessionLength(sessionLength !== 0 ? sessionLength - 1 : 0)}
          handleIncrement={() => setSessionLength(sessionLength + 1)}
        />
      </div>
      <Clock />
      <ClockActions />
    </div>
  );
};

export default ClockContainer;