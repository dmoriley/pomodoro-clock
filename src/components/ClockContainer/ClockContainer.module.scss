.root {
  & > * {
    margin-bottom: var(--l-size);
  }
}

.settings {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
}