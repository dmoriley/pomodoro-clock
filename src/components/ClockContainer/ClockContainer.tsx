import React from "react";
import styles from "./ClockContainer.module.scss";
import { Clock, ClockActions, ClockSetting } from "../index";
import usePomodoro from "../../hooks/pomodoro/pomodoro.hook";

const ClockContainer: React.FC = () => {
  const [
    { power, time, sessionLength, breakLength, status },
    reset,
    playPause,
    sessionDecrement,
    sessionIncrement,
    breakDecrement,
    breakIncrement,
    finishEarly,
  ] = usePomodoro();

  return (
    <div className={styles.root}>
      <div className={styles.settings}>
        {!power ? (
          <React.Fragment>
            <ClockSetting
              title="break length"
              setting={breakLength}
              handleDecrement={breakDecrement}
              handleIncrement={breakIncrement}
            />
            <ClockSetting
              title="session length"
              setting={sessionLength}
              handleDecrement={sessionDecrement}
              handleIncrement={sessionIncrement}
            />
          </React.Fragment>
        ) : (
          <React.Fragment>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 24 24"
              fill="currentColor"
              role="button"
              className="icon icon-l"
              onClick={finishEarly}
            >
              <path d="M5.055 7.06C3.805 6.347 2.25 7.25 2.25 8.69v8.122c0 1.44 1.555 2.343 2.805 1.628L12 14.471v2.34c0 1.44 1.555 2.343 2.805 1.628l7.108-4.061c1.26-.72 1.26-2.536 0-3.256l-7.108-4.061C13.555 6.346 12 7.249 12 8.689v2.34L5.055 7.061Z" />
            </svg>
          </React.Fragment>
        )}
      </div>
      <Clock time={time} status={status} power={power} />
      <ClockActions handleReset={reset} handlePlayPause={playPause} />
    </div>
  );
};

export default ClockContainer;
