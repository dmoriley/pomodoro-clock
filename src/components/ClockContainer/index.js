import ClockContainer, { ClockState } from './ClockContainer';

export default ClockContainer;
export { ClockState };