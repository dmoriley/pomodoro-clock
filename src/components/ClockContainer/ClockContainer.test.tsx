import React from 'react';
import ClockContainer from './ClockContainer';

describe('ClockContainer', () => {
   it('renders without error', () => {

   });
});