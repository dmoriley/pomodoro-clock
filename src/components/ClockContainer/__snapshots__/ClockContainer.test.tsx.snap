// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockContainer renders without error and match snapshot 1`] = `
<div
  class="root"
>
  <div
    class="settings"
  >
    <div
      class="root"
    >
      <h3>
        break length
      </h3>
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-down icon"
          data-icon="arrow-down"
          data-prefix="fas"
          focusable="false"
          role="button"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M374.6 310.6l-160 160C208.4 476.9 200.2 480 192 480s-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 370.8V64c0-17.69 14.33-31.1 31.1-31.1S224 46.31 224 64v306.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0S387.1 298.1 374.6 310.6z"
            fill="currentColor"
          />
        </svg>
         
        <span
          class="setting"
        >
          5
        </span>
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-up icon"
          data-icon="arrow-up"
          data-prefix="fas"
          focusable="false"
          role="button"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="root"
    >
      <h3>
        session length
      </h3>
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-down icon"
          data-icon="arrow-down"
          data-prefix="fas"
          focusable="false"
          role="button"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M374.6 310.6l-160 160C208.4 476.9 200.2 480 192 480s-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 370.8V64c0-17.69 14.33-31.1 31.1-31.1S224 46.31 224 64v306.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0S387.1 298.1 374.6 310.6z"
            fill="currentColor"
          />
        </svg>
         
        <span
          class="setting"
        >
          20
        </span>
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-up icon"
          data-icon="arrow-up"
          data-prefix="fas"
          focusable="false"
          role="button"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="root off"
  >
    <h2>
      Session
    </h2>
    <p
      class="time"
    >
      20:00
    </p>
  </div>
  <div>
    <span
      role="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play icon"
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M361 215C375.3 223.8 384 239.3 384 256C384 272.7 375.3 288.2 361 296.1L73.03 472.1C58.21 482 39.66 482.4 24.52 473.9C9.377 465.4 0 449.4 0 432V80C0 62.64 9.377 46.63 24.52 38.13C39.66 29.64 58.21 29.99 73.03 39.04L361 215z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pause icon"
        data-icon="pause"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M272 63.1l-32 0c-26.51 0-48 21.49-48 47.1v288c0 26.51 21.49 48 48 48L272 448c26.51 0 48-21.49 48-48v-288C320 85.49 298.5 63.1 272 63.1zM80 63.1l-32 0c-26.51 0-48 21.49-48 48v288C0 426.5 21.49 448 48 448l32 0c26.51 0 48-21.49 48-48v-288C128 85.49 106.5 63.1 80 63.1z"
          fill="currentColor"
        />
      </svg>
    </span>
        
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-rotate-right icon"
      data-icon="arrow-rotate-right"
      data-prefix="fas"
      focusable="false"
      role="button"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 48V192c0 17.69-14.31 32-32 32H320c-17.69 0-32-14.31-32-32s14.31-32 32-32h63.39c-29.97-39.7-77.25-63.78-127.6-63.78C167.7 96.22 96 167.9 96 256s71.69 159.8 159.8 159.8c34.88 0 68.03-11.03 95.88-31.94c14.22-10.53 34.22-7.75 44.81 6.375c10.59 14.16 7.75 34.22-6.375 44.81c-39.03 29.28-85.36 44.86-134.2 44.86C132.5 479.9 32 379.4 32 256s100.5-223.9 223.9-223.9c69.15 0 134 32.47 176.1 86.12V48c0-17.69 14.31-32 32-32S496 30.31 496 48z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
